app/models/provider_user.rb

```diff
@@ -12,7 +12,8 @@ class ProviderUser < ActiveRecord::Base
 
   scope :visible_to, ->(provider_user) {
     joins(:provider_permissions)
-      .where(ProviderPermissions.table_name => { provider_id: provider_user.providers.pluck(:id) })
+      .where(ProviderPermissions.table_name => { provider_id: provider_user.provider_permissions.where(manage_users: true).pluck(:provider_id) })
```

Comment:
[There's a scope defined here](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/master/app/models/provider_permissions.rb#L9) which means you might be able to reduce this to:

```ruby
.where(ProviderPermissions.table_name => { provider_id: provider_user.provider_permissions.manage_users.pluck(:provider_id) })
```

I'm not sure about scoping to the permission though. 
[Currently the only use of this particular scope is on the users index](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/ff72db8338eff848b72122db426c8e49803abb01/app/controllers/provider_interface/provider_users_controller.rb#L6). 
It wasn't obvious in the original story whether the current provider user could see all users from visible providers or all users from visible providers where they could manage users.
I don't know what the significance of visibility versus manageability is in this case.

Reply:
Ah, I knew there was something like that somewhere! Will revisit. Maybe we should leave it as it is for now anyway.